### PR TITLE
runtime-rs: Add RTC to QEMU cmdline

### PR DESCRIPTION
**Pull request by emanuellima1**
WiP, still missing implementation on the QemuCmdLine struct

Signed-off-by: Emanuel Lima <emlima@redhat.com>

### CODE REVIEW

File: src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs [hunk 1]
```diff
@@ -1074,6 +1074,54 @@ impl ToQemuParams for DeviceVirtconsole {
...
+
+        true
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for Rtc {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut params = Vec::new();
+        params.push("rtc".to_owned());
```
Comment (pmores) [inline src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1117]: This line should be dropped.  What this function need to output is along the lines of
`-rtc base=utc,driftfix=slew,clock=host`
but as is, the output would apparently be more like
`-rtc rtc,base=utc,driftfix=slew,clock=host`

File: src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs [hunk 2]
```diff
@@ -1074,6 +1074,54 @@ impl ToQemuParams for DeviceVirtconsole {
...
+
+    // Clock is the is the RTC clock driver.
+    clock: String,
+
+    // DriftFix is the drift fixing mechanism.
+    driftfix: String,
+}
+
+impl Rtc {
+    fn new(base: &str, clock: &str, driftfix: &str) -> Rtc {
```
Comment (gkurz) [inline src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1091]: RTC isn't user configurable. The command line we want is exactly `-rtc base=utc,driftfix=slew,clock=host`.
I suggest you drop the arguments to this functions and you just hard-code the values in the `Rtc` instance below.

**Review by gkurz — CHANGES_REQUESTED** [review]
Hi @emanuellima1 ! Thanks for working on this.

The PR currently adds the generation code for `-rtc`. Now you need to have this code called from the `build()` function.

**Comment by gkurz** [comment]
>  Fix conversations

Pushing a fix commit definitely allows to clearly track changes requested by reviewers, but this is pretty much the only advantage. In the end we wan't these fixes to be squashed into the original commit since we just want to record the version that final result, not the intermediary steps induced by the review. And also this will make the `commit message check` happy since it mandates all commits to have a proper SoB tag.

Since you've addressed @pmores 's suggestions, I suggest you squash the `Fix converstations` commit into `runtime-rs: Add RTC to QEMU cmdline`. You should then process my requests and likely squash these changes into the same commit. And then you can force push to the PR.

**Comment by emanuellima1** [comment]
> > Fix conversations
> 
> Pushing a fix commit definitely allows to clearly track changes requested by reviewers, but this is pretty much the only advantage. In the end we wan't these fixes to be squashed into the original commit since we just want to record the version that final result, not the intermediary steps induced by the review. And also this will make the `commit message check` happy since it mandates all commits to have a proper SoB tag.
> 
> Since you've addressed @pmores 's suggestions, I suggest you squash the `Fix converstations` commit into `runtime-rs: Add RTC to QEMU cmdline`. You should then process my requests and likely squash these changes into the same commit. And then you can force push to the PR.

I thought that I could squash when the PR was merged, like I do in Gitlab, but I can do it now too.

**Comment by gkurz** [comment]
> I thought that I could squash when the PR was merged, like I do in Gitlab, but I can do it now too.

This works as well but you'll need all the commits to be properly formatted in order to pass the required commit check CI otherwise the PR won't be mergeable.

Please see https://github.com/kata-containers/kata-containers/actions/runs/8755034747/job/24033016362?pr=9519 .

**Review by gkurz — APPROVED** [review]
LGTM and I could verify that this PR does add ` -rtc base=utc,driftfix=slew,clock=host` on the QEMU command line. Thanks @emanuellima1 !

**Review by Apokleos — APPROVED** [review]
Thx @emanuellima1 for your contribution! 
but no related issue or Fixes #xxx the CI is also happy ? @gkurz 

anyway LGTM !

**Comment by gkurz** [comment]
> Thx @emanuellima1 for your contribution! but no related issue or Fixes #xxx the CI is also happy ? @gkurz
> 

@Apokleos yes, this era is over :tada: . See #9501 .

**Comment by Apokleos** [comment]
> > Thx @emanuellima1 for your contribution! but no related issue or Fixes #xxx the CI is also happy ? @gkurz
> 
> @Apokleos yes, this era is over 🎉 . See #9501 .
>
Yeah, It's over !

**Pull request merged**